frontend/docsettings.lua

```diff
@@ -309,31 +322,22 @@ function DocSettings:purge(sidecar_to_keep)
         end
     end
 
-    local custom_metadata_purged
-    if not sidecar_to_keep then
-        -- custom cover
-        local metadata_file = self:getCoverFile()
-        if metadata_file then
-            os.remove(metadata_file)
-            self:getCoverFile(true) -- reset cache
-            custom_metadata_purged = true
-        end
-        -- custom metadata
-        metadata_file = self:getCustomMetadataFile()
-        if metadata_file then
-            os.remove(metadata_file)
-            custom_metadata_purged = true
-        end
-    end
-
-    if lfs.attributes(self.doc_sidecar_dir, "mode") == "directory" then
-        os.remove(self.doc_sidecar_dir) -- keep parent folders
+    if data_to_purge.custom_cover_file then
+        os.remove(data_to_purge.custom_cover_file)
+        self:getCoverFile(true) -- reset cache
     end
-    if lfs.attributes(self.dir_sidecar_dir, "mode") == "directory" then
-        util.removePath(self.dir_sidecar_dir) -- remove empty parent folders
+    if data_to_purge.custom_metadata_file then
+        os.remove(data_to_purge.custom_metadata_file)
     end
 
-    return custom_metadata_purged
+    if data_to_purge.doc_settings or data_to_purge.custom_cover_file or data_to_purge.custom_metadata_file then
+        if lfs.attributes(self.doc_sidecar_dir, "mode") == "directory" then
+            os.remove(self.doc_sidecar_dir) -- keep parent folders
+        end
+        if lfs.attributes(self.dir_sidecar_dir, "mode") == "directory" then
+            util.removePath(self.dir_sidecar_dir) -- remove empty parent folders
```

Comment:
May be a comment to mention that these will silently do nothing if the directory(ies) is not empty? (If that's the case.)